src/rulesets/Base/mapreduce.jl

```diff
@@ -67,30 +67,55 @@ function rrule(
 end
 
 function rrule(
-    config::RuleConfig{>:HasReverseMode}, ::typeof(sum), f, xs::AbstractArray; dims=:
-)
-    fx_and_pullbacks = map(x->rrule_via_ad(config, f, x), xs)
-    y = sum(first, fx_and_pullbacks; dims=dims)
+    config::RuleConfig{>:HasReverseMode}, ::typeof(sum), f::F, xs::AbstractArray; dims=:
+) where {F}
+    project = ProjectTo(xs)
 
-    pullbacks = last.(fx_and_pullbacks)
+    if _uses_input_only(f, eltype(xs))
+        # Then we can compute the forward pass as usual, save nothing but `xs`:
+        y = sum(f, xs; dims=dims)
+        function sum_pullback_easy(dy)
+            dxs = unthunk(dy) .* conj.(only.(only.(derivatives_given_output.(nothing, f, xs))))
+            return (NoTangent(), NoTangent(), project(dxs))
+        end
+        return y, sum_pullback_easy
+    end
 
-    project = ProjectTo(xs)
+    # In the general case, we need to save all the pullbacks:
+    fx_and_pullbacks = map(x -> rrule_via_ad(config, f, x), xs)
+    y = sum(first, fx_and_pullbacks; dims=dims)
+
+    function sum_pullback_f(dy)
+        # For arrays of arrays, we ought to protect the element against broadcasting:
+        dys = dims isa Colon ? Ref(unthunk(dy)) : unthunk(dy)
```

Comment:
>=  performant to use 1-tuple rather than `Ref`
```suggestion
        dys = dims isa Colon ? (unthunk(dy),) : unthunk(dy)
```

Reply:
Isn't `Ref` the standard thing, and what's used internally? I think this makes the intention a little clearer. 
```
julia> Broadcast.broadcastable(:x)
Base.RefValue{Symbol}(:x)

julia> Broadcast.broadcastable(sin)
Base.RefValue{typeof(sin)}(sin)
```